examples/golang/docker-compose-go.yaml

```diff
@@ -7,7 +7,8 @@ services:
     command: >
       bash -c "
       bazel build  //examples/golang/simple:simple.so
-      && cp -a ./bazel-bin/examples/golang/simple/simple.so /output"
+      && cp -a ./bazel-bin/examples/golang/simple/simple.so /output
+      && bazel clean --expunge"
```

Comment:
we dont want to do this i think - this will cane the users cache - what was the reason for adding this ?

Reply:
I have tried to clean compile image of Go plugin, but I found that the disk pressure is still heavy([92%](https://dev.azure.com/cncf/envoy/_build/results?buildId=127410&view=logs&j=f1a49366-51c0-540f-0294-f73c2bb311a9&t=7fd036c9-61ec-5b59-3645-e90950dc67a6&l=13897)) and `skywalking-ui` [failed again](https://dev.azure.com/cncf/envoy/_build/results?buildId=127410&view=logs&j=f1a49366-51c0-540f-0294-f73c2bb311a9&t=7fd036c9-61ec-5b59-3645-e90950dc67a6&l=13891).
So I started looking for any disk issues caused by the Go sandbox, and found that the output of bazel would occupy more disks, so I tried the [commit](https://github.com/envoyproxy/envoy/pull/24848/commits/79daf1ef83a707da92d42926152877d1d1425fa0) and it can work normally.

Reply:
either way we cant do this to end users (who may/not have their real bazel cache pointed here)

i am testing the current solutions to the (now ~confirmed) disk pressure issue #25399 

my first test which was to just run `docker image prune -f` after each sandbox invokation (ie just clean up intermediate build images) seems to fix the problem by itself, even without actually removing the build image - which as it used again would be better to avoid

im also going to test the other solution - making elasticsearch less disk greedy to see if that resolves

Reply:
OK. I'll revert it and merge upstream after #25374 addresed.

Reply:
updated